sdk/identity/Azure.Identity/tests/VisualStudioCredentialLiveTests.cs

```diff
@@ -0,0 +1,186 @@
+            var testProcess = new TestProcess { Timeout = 10000 };
+            testProcess.Started += (o, e) => cts.Cancel();
+
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, fileSystem, new TestProcessService(testProcess)));
+            Assert.CatchAsync<OperationCanceledException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), cts.Token));
+        }
+
+        [Test]
+        public void AuthenticateWithVsCredential_NoVsInstalled()
+        {
+            var fileSystem = CreateTestFileSystem();
+            fileSystem.FileExistsHandler = p => false;
+
+            var (_, _, processOutput) = CreateTestToken();
+            var testProcess = new TestProcess { Output = processOutput };
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, fileSystem, new TestProcessService(testProcess)));
+            Assert.ThrowsAsync<CredentialUnavailableException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), CancellationToken.None));
+        }
+
+        [Test]
+        public void AuthenticateWithVsCredential_NoJsonFileFound()
+        {
+            var (_, _, processOutput) = CreateTestToken();
+            var testProcess = new TestProcess { Output = processOutput };
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, new TestFileSystemService(), new TestProcessService(testProcess)));
+            Assert.ThrowsAsync<CredentialUnavailableException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), CancellationToken.None));
+        }
+
+        [Test]
+        public void AuthenticateWithVsCredential_BrokenJsonFileFound()
+        {
+            var (_, _, processOutput) = CreateTestToken();
+            var testProcess = new TestProcess { Output = processOutput };
+            var fileSystem = new TestFileSystemService { ReadAllHandler = p => "{\"Some\": false}" };
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, fileSystem, new TestProcessService(testProcess)));
+            Assert.ThrowsAsync<AuthenticationFailedException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), CancellationToken.None));
+        }
+
+        [Test]
+        public void AuthenticateWithVsCredential_ProcessFailed()
+        {
+            var testProcess = new TestProcess { Error = "Some error" };
+            var fileSystem = CreateTestFileSystem();
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, fileSystem, new TestProcessService(testProcess)));
+
+            Assert.ThrowsAsync<AuthenticationFailedException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), CancellationToken.None));
```

Comment:
Same here, `CredentialUnavailableException`, if process failed we should look for another credentia.

Reply:
Here I disagree, if the process failed I think we should notify the user rather than continuing to try other credentials. If we don't stop we risk the user executing with an account they didn't intend. The only way to do this is to raise AuthenticationFailedExcpetion. CredentialUnavailableException is intended to be thrown in cases were authentication wouldn't be expected to work. In the case of the VS credential I would expect this to be when VS isn't installed or no user has logged in via VS.  Any unexpected failure should halt the authentication.